packages/api/src/web3Middleware/index.ts

```diff
@@ -34,17 +35,24 @@ export class Web3Middleware {
   }
 
   async updateAddress (feedInfo: FeedInfo) {
-    const contractAddress = await this.getContractAddress(feedInfo)
+    const contractInfo = await this.getContractAddress(feedInfo)
+    console.log('contractInfo in updateAddress', contractInfo)
     const feed = this.repositories.feedRepository.get(feedInfo.feedFullName)
 
-    if (feed && contractAddress && contractAddress !== feed.address) {
+    if (
+      feed &&
+      contractInfo &&
+      contractInfo.contractAddress &&
+      contractInfo.contractAddress !== feed.address
```

Comment:
We can use the optional chaining operator
```suggestion
      contractInfo.contractAddress &&
      contractInfo?.contractAddress !== feed?.address
```